src/App.jsx

```diff
@@ -1,248 +1,362 @@
-                  type="button"
-                  className="delete"
-                />
-              </span>
-            </p>
-          </div>
-
-          <div className="panel-block is-flex-wrap-wrap">
-            <a
-              href="#/"
-              data-cy="AllCategories"
-              className="button is-success mr-6 is-outlined"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1 is-info"
-              href="#/"
-            >
-              Category 1
-            </a>
-
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1"
-              href="#/"
-            >
-              Category 2
-            </a>
-
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1 is-info"
-              href="#/"
-            >
-              Category 3
-            </a>
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1"
-              href="#/"
-            >
-              Category 4
-            </a>
-          </div>
-
-          <div className="panel-block">
-            <a
-              data-cy="ResetAllButton"
-              href="#/"
-              className="button is-link is-outlined is-fullwidth"
-            >
-              Reset all filters
-            </a>
-          </div>
-        </nav>
-      </div>
-
-      <div className="box table-container">
-        <p data-cy="NoMatchingMessage">
-          No products matching selected criteria
-        </p>
-
-        <table
-          data-cy="ProductTable"
-          className="table is-striped is-narrow is-fullwidth"
-        >
-          <thead>
-            <tr>
-              <th>
-                <span className="is-flex is-flex-wrap-nowrap">
-                  ID
-
-                  <a href="#/">
-                    <span className="icon">
-                      <i data-cy="SortIcon" className="fas fa-sort" />
-                    </span>
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+
+const products = productsFromServer.map((product) => {
+  const category = categoriesFromServer
+    .find(categoryData => categoryData.id === product.categoryId);
+
+  const user = usersFromServer
+    .find(userData => userData.id === category.ownerId);
+
+  return {
+    ...product,
+    category,
+    user,
+  };
+});
+
+const users = [{ id: 0, name: 'All' }, ...usersFromServer];
+
+const getPreparedProducts = (
+  productsData,
+  owner,
+  query,
+  selectedCategories,
+) => {
+  let resultData = [...productsData];
+
+  if (owner) {
+    if (owner === 'All') {
+      resultData = [...products];
+    } else {
+      resultData = resultData.filter(({ user }) => user.name === owner);
+    }
+  }
+
+  if (query) {
+    resultData = resultData
```

Comment:
Would be better to to change `resultData` to more appropriate variable. Because it's not obvious that is a products array